src/auth/feedback-secret.guard.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  Injectable,
+  CanActivate,
+  ExecutionContext,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Env } from 'src/env';
+
+@Injectable()
+export class FeedbackSecretGuard implements CanActivate {
+  constructor() {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const request = context.switchToHttp().getRequest();
+    const secret = Env.SECRET.FEEDBACK;
+    const requestSecret = request.headers['x-feedback-secret'];
+
+    if (requestSecret && requestSecret === secret) {
+      return true;
+    } else {
+      throw new UnauthorizedException('Invalid feedback secret');
```

Comment:
PullProofAI

There is a potential security vulnerability in the `FeedbackSecretGuard` class. The comparison of the request secret with the feedback secret is done using a simple equality check, which can be susceptible to timing attacks. To mitigate this issue, it's recommended to use a constant-time comparison method for comparing secrets.

```diff
- if (requestSecret && requestSecret === secret) {
+ if (requestSecret && crypto.timingSafeEqual(Buffer.from(requestSecret), Buffer.from(secret))) {
```

This change ensures that the comparison is done in constant time, making it resistant to timing attacks and improving the security of the authentication process.



<!-- This is an auto-generated comment by PullProofAI -->